File: JavaToKotlin/apps(kotlin)/src/main/kotlin/io/dustin/apps/board/domain/blockeduser/service/WriteBlockedUserService.java
package io.dustin.apps.board.domain.blockeduser.service;

import io.dustin.apps.board.domain.blockeduser.model.BlockedUser;
import io.dustin.apps.board.domain.blockeduser.repository.BlockedUserRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

@RequiredArgsConstructor
@Service
public class WriteBlockedUserService {

    private final BlockedUserRepository blockedUserRepository;

    public BlockedUser create(Long fromUserId, Long toUserId){
        BlockedUser blockedUser = BlockedUser.builder()
                    .fromUserId(fromUserId)
                    .toUserId(toUserId)
                    .build();
        this.blockedUserRepository.save(blockedUser);
        return blockedUser;
    }

    @Transactional
    public void delete(Long fromUserId, Long toUserId) { blockedUserRepository.deleteByFromUserIdAndToUserId(fromUserId, toUserId);}



}

File: apps/src/main/kotlin/io/dustin/apps/board/domain/blockeduser/service/WriteBlockedUserService.kt
package io.dustin.apps.board.domain.blockeduser.service

import io.dustin.apps.board.domain.blockeduser.model.BlockedUser
import io.dustin.apps.board.domain.blockeduser.repository.BlockedUserRepository
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Service
class WriteBlockedUserService (
    private val blockedUserRepository: BlockedUserRepository,
){

    /**
     * 유저 차단
     */
    fun create(fromUserId: Long, toUserId: Long): BlockedUser {
        val blockedUser = BlockedUser(
            fromUserId = fromUserId,
            toUserId = toUserId
        )
        blockedUserRepository.save(blockedUser)
        return blockedUser
    }

    /**
     * 유저 차단 해제
     */
    fun delete(fromUserId: Long, toUserId: Long) {
        blockedUserRepository.deleteByFromUserIdAndToUserId(fromUserId, toUserId)
    }

}
